test/integration/virtual-rules/index.js

```diff
@@ -0,0 +1,315 @@
+describe('virtual rules', function() {
```

Comment:
I think this file should be split up, give each rule its own .js file. I'm afraid this is going to get messy quickly if we don't.